### PR TITLE
Initial GKL integration

### PR DESCRIPTION
**Pull request by gspowley**
**Initial integration of GKL**
* Removed native build related items from `build.gradle`
* Removed native code from src tree
* Refactored `PairHMM.java` and `VectorLoglessPairHMM.java` to use GKL
* Updated `VectorPairHMMUnitTest.java` to use GKL
* Added integration tests to `IntelDeflaterIntegrationTest.java`

**Notes**
* PairHMM has been tested in HaplotypeCaller and GVCF output is md5sum equivalent to the PairHMM currently in GATK
* PairHMM in GKL is still single threaded, but about ***1.4x faster*** than existing PairHMM, due to fixing a performance issue in the native code
* Next steps are captured in #1903 #1946

### CODE REVIEW

**Comment by gspowley** [comment]
Thinking more about it, we could merge this PR as is. PairHMM and HaplotypeCaller will continue to work as they currently do (only faster). We can hook up the PairHMM arguments later.

For IntelDeflater, we need a new release of HTSJDK, so we can make a new PR after that.

@akiezun @droazen @lbergelson Let me know how you want to move forward.

File: src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java [hunk 1]
```diff
@@ -1,29 +1,129 @@
...
+                inflater.end();
+            } catch (java.util.zip.DataFormatException e) {
+                e.printStackTrace();
+            }
+
+            Assert.assertEquals(input, result);
+        }
     }
 
+    @Test(enabled = false)
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java]: why is this test disabled?

**Comment by akiezun** [comment]
I can grab this one - i'll test it and measure

File: src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java (hunk 1 above)
Comment (gspowley) [inline src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java]: The test uses `SAMFileWriterFactory.setDeflaterFactory`, which isn't available in a released version of HTSJDK.

**Comment by akiezun** [comment]
@gspowley this works for me - i tested on linux and mac. I do see speedups too!

I think it's good to me merged. Can you enter tickets for the remaining work items and add them as comments here? 

mac osx 
master
real	1m51.375s
user	2m39.498s

real	1m51.152s
user	2m38.079s

new gkl
real	1m46.342s
user	2m38.097s

real	1m46.603s
user	2m36.670s

linux (cloud vm)
master
JNI 73.7
real	2m35.924s
user	3m37.316s

JNI 74.39
real	2m32.028s
user	3m29.936s


new gkl
JNI 57.67
real	2m20.550s
user	3m27.520s

JNI 53.92241665
real	2m18.047s
user	3m20.288s

**Comment by akiezun** [comment]
assigning to @gspowley to enter remaining tickets.

**Comment by gspowley** [comment]
Use these issues to cover next integration steps. #1903 #1946

File: src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java [hunk 2]
```diff
@@ -1,29 +1,129 @@
...
+    private final static Logger log = LogManager.getLogger(IntelDeflaterIntegrationTest.class);
+    private static final String publicTestDirRelative = "src/test/resources/";
+    private static final String publicTestDir = new File(gatkDirectory, publicTestDirRelative).getAbsolutePath() + "/";
+    private static final String INPUT_FILE = publicTestDir + "CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam";
 
-        if ( NativeUtils.runningOnPPCArchitecture() ) {
-            throw new SkipException("IntelDeflater not available for this architecture");
-        }
+    @Test(enabled = true)
+    public void loadIntelDeflater() {
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java]: Can you restore the `SkipExceptions` when `! NativeUtils.runningOnLinux()` or `NativeUtils.runningOnPPCArchitecture()`? We want this test to be skipped on unsupported architectures/platforms.

File: src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java [hunk 3]
```diff
@@ -1,29 +1,129 @@
...
 public class IntelDeflaterIntegrationTest extends BaseTest {
 
-    @Test
-    public void testIntelDeflaterIsAvailable(){
-        if ( ! NativeUtils.runningOnLinux() ) {
-            throw new SkipException("IntelDeflater not available on this platform");
-        }
+    private final static Logger log = LogManager.getLogger(IntelDeflaterIntegrationTest.class);
+    private static final String publicTestDirRelative = "src/test/resources/";
+    private static final String publicTestDir = new File(gatkDirectory, publicTestDirRelative).getAbsolutePath() + "/";
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java]: Use the declarations in `BaseTest` for these paths.

File: src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java (hunk 2 above)
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java]: Also, decouple the tests that merely test loading vs. those that test functionality.

**Comment by akiezun** [comment]
back to @gspowley to address comments

**Comment by gspowley** [comment]
Updated tests to address comments from @droazen.

**Pull request merged**